src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -237,13 +250,14 @@ export class CellExecution {
         traceCellMessage(this.cell, 'Completed with errors');
         this.sendPerceivedCellExecute();
         if (!this.isEmptyCodeCell) {
-            await chainWithPendingUpdates(this.editor.document, (edit) => {
+            await chainWithPendingUpdates(this.cell.notebook, (edit) => {
                 traceCellMessage(this.cell, 'Update run run duration');
                 const metadata = this.cell.metadata.with({ lastRunDuration: this.stopWatch.elapsedTime });
-                edit.replaceNotebookCellMetadata(this.editor.document.uri, this.cell.index, metadata);
+                edit.replaceNotebookCellMetadata(this.cell.notebook.uri, this.cell.index, metadata);
```

Comment:
This was one that bit me when I was working on my toy notebook implementation. Was passing in the document uri to my update functions instead of the notebook uri.

Reply:
Yup, i found one place i did that as well.